src/Nethermind/Nethermind.Core/Eip2930/AccessList.cs

```diff
@@ -1,30 +1,109 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
+using System;
 using System.Collections.Generic;
 using Nethermind.Int256;
 
-namespace Nethermind.Core.Eip2930
+namespace Nethermind.Core.Eip2930;
+
+public abstract record AccessListItem
 {
-    public class AccessList
+    public record Address(Core.Address Value) : AccessListItem;
+
+    public record StorageKey(UInt256 Value) : AccessListItem;
+}
+
+public class AccessList
+{
+    private readonly List<AccessListItem> _items;
+
+    private AccessList(List<AccessListItem> items)
+    {
+        _items = items;
+    }
+
+    public static AccessList Empty() => new AccessList(new List<AccessListItem>());
+
+    public IReadOnlyCollection<AccessListItem> Raw => _items;
```

Comment:
Every place where `Raw` is used, `AsEnumerable` would do better, maybe with exception of `AccessListCost`, though, Maybe would be better to have a method `Cost` on the access list itself and just drop the `Raw` property as it gives insight to internal structure (breaks encapsulation)